node/modules/lp2p/pubsub.go

```diff
@@ -27,17 +29,47 @@ func init() {
 	pubsub.GossipSubDlazy = 12
 	pubsub.GossipSubDirectConnectInitialDelay = 30 * time.Second
 }
+func ScoreKeeper() *dtypes.ScoreKeeper {
+	return new(dtypes.ScoreKeeper)
+}
+
+type GossipIn struct {
+	fx.In
+	Mctx helpers.MetricsCtx
+	Lc   fx.Lifecycle
+	Host host.Host
+	Nn   dtypes.NetworkName
+	Bp   dtypes.BootstrapPeers
+	Cfg  *config.Pubsub
+	Sk   *dtypes.ScoreKeeper
+}
+
+func getDrandTopic() (string, error) {
+	var drandInfo = struct {
+		Hash string `json:"hash"`
+	}{}
+	err := json.Unmarshal([]byte(build.DrandChain), &drandInfo)
+	if err != nil {
+		return "", xerrors.Errorf("could not unmarshal drand chain info: %w", err)
+	}
+	return "/drand/pubsub/v0.0.0/" + drandInfo.Hash, nil
+}
 
-func GossipSub(mctx helpers.MetricsCtx, lc fx.Lifecycle, host host.Host, nn dtypes.NetworkName, bp dtypes.BootstrapPeers, cfg *config.Pubsub) (service *pubsub.PubSub, err error) {
+func GossipSub(in GossipIn) (service *pubsub.PubSub, err error) {
 	bootstrappers := make(map[peer.ID]struct{})
-	for _, pi := range bp {
+	for _, pi := range in.Bp {
 		bootstrappers[pi.ID] = struct{}{}
 	}
-	isBootstrapNode := cfg.Bootstrapper
+	isBootstrapNode := in.Cfg.Bootstrapper
```

Comment:
We also have to include the drand bootstrap peers in this list so that PX can work.